models/shreds.py

```diff
@@ -63,36 +63,35 @@ class ClusterMember(EmbeddedDocument):
     angle = FloatField()
 
 
-class Cluster(EmbeddedDocument):
+class Cluster(Document):
     """Cluster of one or more shreds.
 
-    Clusters are immutable once created.
-
     Shred membership described with a ClusterMember embedded document, which
     contains a reference to the shred and its relative position and angle in a
-    cluster."""
-    id = StringField(required=True, max_length=36, primary_key=True)
-    members = ListField(EmbeddedDocumentField(ClusterMember))
-    parents = ListField(ReferenceField('Cluster'))
-
-    @property
-    def features(self):
-        # TODO: persist features on creation.
-        return self.members[0].shred.features
+    cluster.
 
-
-class Taggable(Document):
+    Cluster also contains user-generated tagging results (tags field).
+    """
     id = StringField(max_length=200, default='', primary_key=True)
-    object = EmbeddedDocumentField(Cluster)
+
     users_count = IntField(default=0, db_field='usersCount')
     users_skipped = ListField(ReferenceField(User), db_field='usersSkipped')
     users_processed = ListField(ReferenceField(User),
                                 db_field='usersProcessed')
+
     batch = StringField()
     tags = ListField(EmbeddedDocumentField(ShredTags))
 
+    members = ListField(EmbeddedDocumentField(ClusterMember))
+    parents = ListField(ReferenceField('Cluster'))
```

Comment:
Parents will contain pointers to two (or zero for base clusters) clusters. Members will be a flat list of all shred with their positions.